core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -857,6 +858,364 @@ KOKKOS_INLINE_FUNCTION_DELETED Impl::ThreadVectorRangeBoundariesStruct<
+  explicit NoReductionTag() = default;
+};
+
+inline constexpr NoReductionTag no_reduction_tag{};
+
+template <MDTeamRangeLastRank LR, MDTeamRangeParThread PT,
+          MDTeamRangeParVector PV>
+struct MDTeamRangeStatus {
+  static constexpr MDTeamRangeLastRank last_rank   = LR;
+  static constexpr MDTeamRangeParThread par_thread = PT;
+  static constexpr MDTeamRangeParVector par_vector = PV;
+};
+
+template <int CurrentRank, int TotalRank, int ParThreadRank, int ParVectorRank,
+          Iterate Iter>
+struct MDTeamRangeNextStatus {
+  using rank_type                            = int;
+  static constexpr rank_type current_rank    = CurrentRank;
+  static constexpr rank_type total_rank      = TotalRank;
+  static constexpr rank_type par_thread_rank = ParThreadRank;
+  static constexpr rank_type par_vector_rank = ParVectorRank;
+  static constexpr Iterate iter              = Iter;
+
+  // We have to recursively process ranks [0..total_rank-1]
+  using type = MDTeamRangeStatus<
+      (iter == Iterate::Right)
+          ? static_cast<MDTeamRangeLastRank>(current_rank == total_rank)
+          : static_cast<MDTeamRangeLastRank>(current_rank == -1),
+      static_cast<MDTeamRangeParThread>(current_rank == par_thread_rank),
+      static_cast<MDTeamRangeParVector>(current_rank == par_vector_rank)>;
+};
+
+template <int CurrentRank, int TotalRank, int ParThreadRank, int ParVectorRank,
+          Iterate Iter>
+using MDTeamRangeStatus_t =
+    typename MDTeamRangeNextStatus<CurrentRank, TotalRank, ParThreadRank,
+                                   ParVectorRank, Iter>::type;
+
+template <int totalRank, Iterate Iter, typename ExecSpace,
+          MDTeamRangeThreadAndVector t_and_v>
+struct ParallelRank;
+
+// FOR SERIAL ONLY
+template <int TotalRank, MDTeamRangeThreadAndVector t_and_v>
+struct ParallelRank<TotalRank, Iterate::Left, DefaultHostExecutionSpace,
+                    t_and_v> {
+  // For threads deploy thread parallelism on Right most (i.e. slowed index)
+  static constexpr int par_rt  = TotalRank - 1;
+  static constexpr int par_rv  = 0;
+  static constexpr int invalid = -2;
+};
+template <int TotalRank, MDTeamRangeThreadAndVector t_and_v>
+struct ParallelRank<TotalRank, Iterate::Right, DefaultHostExecutionSpace,
+                    t_and_v> {
+  // For threads deploy thread parallelism on left most (i.e. slowed index)
+  static constexpr int par_rt  = 0;
+  static constexpr int par_rv  = TotalRank - 1;
+  static constexpr int invalid = -2;
+};
+
+// template <typename iType, class Closure, class Member>
+// void parallel_for(Kokkos::TeamThreadRange const&, Closure const&);
+
+template <typename TeamHandle>
+KOKKOS_INLINE_FUNCTION auto nested_policy(
+    MDTeamRangeStatus<MDTeamRangeLastRank::NotLastRank,
+                      MDTeamRangeParThread::NotParThread,
+                      MDTeamRangeParVector::NotParVector>
+        status,
+    TeamHandle const& team, int count) {
+  return RangePolicy<>(0, count);
```

Comment:
This can't work right now. Eventually maybe yes if we have an "inline execution space" which is usable inside of parallel contexts, but right now this won't work.

Reply:
Fixed in [4d802ec](https://github.com/kokkos/kokkos/pull/5238/commits/4d802ec0db4591eda75b5565b0042ef04763c875)